internal/maintenance/periodic_job_enqueuer.go

```diff
@@ -243,7 +243,10 @@ func (s *PeriodicJobEnqueuer) Remove(periodicJobHandle rivertype.PeriodicJobHand
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	delete(s.periodicJobs, periodicJobHandle)
+	if periodicJob, ok := s.periodicJobs[periodicJobHandle]; ok {
+		delete(s.periodicJobs, periodicJobHandle)
+		delete(s.periodicJobIDs, periodicJob.ID)
```

Comment:
Although it creates some indirection, this is the kind of scenario with needing to update multiple things upon deletion is one where I'd typically extract these into a shared function. This guarantees the two call sites use the same logic for manipulating these maps. Otherwise I fear drift if one is updated and the other is overlooked. Something like:

```go

func (s *PeriodicJobEnqueuer) removeLocked(periodicJobHandle rivertype.PeriodicJobHandle) {
// both deletes
}
```

Which you can then call in either location. The compiler will probably inline it and IMO the reduced risk of concurrency bugs outweighs the small cost of indirection.

Your call though! The tests at least provide us some protection.